### ISSUE TITLE
Gear Information appears blank until a change is made

### ISSUE THREAD

**Issue by phossystems** [NicoSchlueter/HelicalGear#3]
Gear Information appears blank until a change is made. This does not happen when opening the dialog for the first time.

<img width="270" alt="screenshot 2019-01-09 at 15 31 48" src="https://user-images.githubusercontent.com/30301307/50905703-b2577e00-1423-11e9-948a-c59d4be53090.png">

**Comment by lhns** [comment]
Hi!
I've just submitted a few PRs including one which should fix this issue.
The code reformatting PR might be a bit controversial but in my opinion it makes the code a bit easier to read and modify.
Let me know what you think.

**Comment by phossystems** [comment]
Great Work!
I kept the code as close to original as possible, but since Ross told me he wasn't interested in merging this into the app-store version we might as well make it a bit easier to read. I also cleaned up my annotations a bit.
I found a minor bug with the stat fix. I will open up new issues for that and the BaseFeature incompatibility. I hope to find some time to work on this some more soon.
Thank you for your contribution!
